test/generate-missing-examples.test.ts

```diff
@@ -90,4 +91,77 @@ test('test end-to-end and translation to Python', async () => {
   } finally {
     await assembly.cleanup();
   }
-});
\ No newline at end of file
+});
+
+test('test end-to-end and translation to Python with compressed assembly', async () => {
+  const assembly = await AssemblyFixture.fromSource(
+    {
+      'index.ts': `
+      export interface MyClassProps {
+        readonly someString: string;
+        readonly someNumber: number;
+      }
+
+      export class MyClass {
+        constructor(value: string, props: MyClassProps) {
+          Array.isArray(value);
+          Array.isArray(props);
+        }
+      }
+    `,
+    },
+    {
+      name: 'my_assembly',
+      jsii: DUMMY_ASSEMBLY_TARGETS,
+    },
+    {
+      compressAssembly: true,
+    },
+  );
+  try {
+    // ensure the compressed assembly exists and the file at SPEC_FILE_NAME is a redirect schema
+    expect(fs.existsSync(path.join(assembly.directory, SPEC_FILE_NAME_COMPRESSED))).toBeTruthy();
+    const schema = await fs.readJson(path.join(assembly.directory, SPEC_FILE_NAME), { encoding: 'utf-8' });
+    expect(schema).toEqual({
+      schema: 'jsii/file-redirect',
+      compression: 'gzip',
+      filename: SPEC_FILE_NAME_COMPRESSED,
+    });
+
+    const outputTablet = path.join(assembly.directory, 'test.tbl.json');
+
+    await generateMissingExamples([assembly.directory], {
+      extractOptions: {
+        cache: outputTablet,
+      },
+    });
+
+    // ensure the file at SPEC_FILE_NAME is still a redirect schema
+    expect(await fs.readJson(path.join(assembly.directory, SPEC_FILE_NAME), { encoding: 'utf-8' })).toEqual({
+      schema: 'jsii/file-redirect',
+      compression: 'gzip',
+      filename: SPEC_FILE_NAME_COMPRESSED,
+    });
```

Comment:
This is the key. I believe this assertion demonstrates that the pipeline build workflow will no longer write over the redirect schema at `.jsii`.

Reply:
To make sure I understand correctly, in the previous workflow, the `replaceAssembly` implementation was overwriting the `.jsii` file. But in the case where a compressed assembly is used, we need it to actually replace the `.jsii.gz` file. Is that right?

Reply:
yes. `writeAssembly` should take that into account. It checks for `.jsii.gz` and if it finds it, replaces that instead.